File: lib/src/main/java/bruto/variability/ArgumentVariability.java
package bruto.variability;

import java.lang.reflect.Type;

/**
 * describes the range of values which might be assumed by a given parameter
 */
public abstract class ArgumentVariability {

    public abstract boolean matches(Type aType);

    public abstract ArgumentVariabilityWalker newWalker(Type aType);

}

File: lib/src/main/scala/bruto/variability/ArgumentVariability.scala
package bruto.variability

import java.lang.reflect.Type

/**
 * describes the range of values which might be assumed by a given parameter
 */
abstract class ArgumentVariability {

  def matches(aType : Type) : Boolean

  def newWalker(aType : Type) : ArgumentVariabilityWalker

}
